src/ringct/rctSigs.cpp

```diff
@@ -337,7 +361,8 @@ namespace rct {
 
       std::stringstream ss;
       binary_archive<true> ba(ss);
-      const size_t inputs = rv.pseudoOuts.size();
+      CHECK_AND_ASSERT_THROW_MES(!rv.mixRing.empty(), "Empty mixRing");
+      const size_t inputs = is_simple(rv.type) ? is_bulletproof(rv.type) ? rv.p.pseudoOuts.size() : rv.pseudoOuts.size() : rv.mixRing[0].size();
```

Comment:
Is this change necessary at all? The variable `inputs` is only relevant for serializing pre-Bulletproof pseudoOuts, so the existing code `inputs = rv.pseudoOuts.size();` seems to just work (i.e. with Bulletproof+Simple, it's ignored in `serialize_rctsig_base` anyway, and with pre-BP/BP+Full, it won't matter either).

Reply:
Yes, because relying on the knowledge that the other piece of code won't actually use the parameter in some subset of cases is brittle.

Reply:
OK.

Reply:
Using pseudoOuts at all feels wrong somehow. Couldn't you just check is_simple and use mixRing.size()/mixRing[0].size()?

Reply:
True, this would simplify a bit, I'll change it.